ray-operator/controllers/ray/common/service.go

```diff
@@ -62,12 +62,11 @@ func BuildServiceForHeadPod(ctx context.Context, cluster rayv1.RayCluster, label
 	default_namespace := cluster.Namespace
 	default_type := cluster.Spec.HeadGroupSpec.ServiceType
 
-	defaultAppProtocol := utils.DefaultServiceAppProtocol
-	// `ports_int` is a map of port names to port numbers, while `ports` is a list of ServicePort objects
-	ports_int := getServicePorts(cluster)
+	// `serviceMapPortInfo` is a map of port names to port numbers, while `ports` is a list of ServicePort objects
+	serviceMapPortInfo := getServicePorts(cluster)
 	ports := []corev1.ServicePort{}
-	for name, port := range ports_int {
-		svcPort := corev1.ServicePort{Name: name, Port: port, AppProtocol: &defaultAppProtocol}
+	for svc, portInfo := range serviceMapPortInfo {
+		svcPort := corev1.ServicePort{Name: svc, Port: portInfo.PortNumber, AppProtocol: &portInfo.Protocol}
```

Comment:
Is it possible that the `&portInfo.Protocol` here points to en empty value? I think in that case we should still use the `&defaultAppProtocol`

Reply:
fixed